cdap-runtime-ext-remote-hadoop/src/main/java/io/cdap/cdap/runtime/spi/provisioner/remote/RemoteHadoopProvisioner.java

```diff
@@ -119,7 +126,27 @@ public Cluster createCluster(ProvisionerContext context) {
       properties.put(ClusterProperties.KERBEROS_KEYTAB, keytab);
     }
 
-    return new Cluster(conf.getHost(), ClusterStatus.RUNNING, nodes, properties);
+    return new Cluster(host, ClusterStatus.RUNNING, nodes, properties);
+  }
+
+  String selectEdgeNode(String hostConfigValue, String profileName) {
+    String host;
+    LOG.debug("Retrieved edge nodes: " + hostConfigValue + " for profile: " + profileName);
+    if (hostConfigValue.contains(HOSTS_SEPARATOR)) {
+      List<String> hosts = Arrays.stream(hostConfigValue.split(HOSTS_SEPARATOR))
+        .map(String::trim).collect(Collectors.toList());
```

Comment:
I suppose that some cases like `host1,,host2`  should be handled by filtering not valid values, as we are not validating this field on profile edit.

Reply:
technically, if the value are correct is not checked in the original code, same as here.  Like: host1, my poem, host2

Reply:
Yes, but in default implementation it requires field host to be filled, so it will be avoid to be empty. With this feature it is possible to add host line like `host1,,host2` and it will be interpreted like 3 hosts, one of them would be empty...